notebooks/optimization.ipynb

```diff
@@ -0,0 +1,388 @@
+    "    ref_sld = Parameter(2, 'Reference layer SLD', (-1.9, 9.4))\n",
+    "    \n",
+    "    # Tell HOGBEN that these parameters should be optimized\n",
+    "    ref_thick.optimize = True\n",
+    "    ref_sld.optimize = True\n",
+    "    solvent1_sld.optimize = True\n",
+    "    \n",
+    "    # Construct the layers\n",
+    "    solvent1 = SLD(solvent1_sld, name='Solvent 1')(rough=3)\n",
+    "    solvent2 = SLD(solvent2_sld, name='Solvent 2')(rough=3)    \n",
+    "    layer1 = SLD(layer1_sld, name=\"Layer 1\")(thick=layer1_thick, rough=layer1_rough, vfsolv=vfsolv1)\n",
+    "    layer2 = SLD(layer2_sld, name=\"Layer 2\")(thick=layer2_thick, rough=layer2_rough, vfsolv=vfsolv2)\n",
+    "    layer3 = SLD(layer3_sld, name=\"Layer 3\")(thick=layer3_thick, rough=layer3_rough, vfsolv=vfsolv3)\n",
+    "    ref_layer = SLD(ref_sld, name=\"Reference Layer\")(thick=ref_thick, rough=3)    \n",
+    "    substrate = SLD(2.074, name='Substrate')(rough=3)\n",
+    "\n",
+    "    # Put all fitting parameters in a list\n",
+    "    params = [\n",
+    "        vfsolv1,\n",
+    "        vfsolv2,\n",
+    "        vfsolv3,\n",
+    "        layer1_sld,\n",
+    "        layer2_sld,\n",
+    "        layer3_sld,\n",
+    "    ]\n",
+    "    \n",
+    "    # Set all fitting parameters to be varying\n",
+    "    for param in params:\n",
+    "        param.vary = True\n",
+    "    \n",
+    "    # Create a structure, separating each layer with a `|`\n",
+    "    sample1 = substrate | ref_layer | layer1 | layer2 | layer3 | solvent1\n",
+    "    sample2 = substrate | ref_layer | layer1 | layer2 | layer3 | solvent2\n",
+    "    \n",
+    "    return [sample1, sample2]"
+   ],
+   "id": "d7df5e2cd043e5a4",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "Now just as we did previously, we simply put the above structure in a HOGBEN sample using `Sample`. Then we need to define the `angle_times` object and tell HOGBEN to optimise the parameters.\n",
+   "id": "66aa068983ec9b49"
+  },
+  {
+   "metadata": {},
+   "cell_type": "code",
+   "source": [
+    "structures = solvent_ref_layer()\n",
+    "sample = Sample(structures)\n",
+    "angle_times = [(0.7, 100, 100),\n",
+    "               (2.3, 100, 400),\n",
+    "               ]    \n",
+    "optimise_parameters(sample, angle_times)\n"
+   ],
+   "id": "7df0f1ad78e9753a",
+   "outputs": [],
+   "execution_count": null
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 2
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython2",
+   "version": "2.7.6"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
```

Comment:
In this notebook, would it be possible to add how to use different angle_times for each contrast, and how to use a different background etc?
It is currently not clear where these should be fed in.

Reply:
I'll add that now. Apologies for being late, I was expecting an email notification with the requested changes but didn't get that. Will also add another unit tests to check the order, just added another few as well.